Geometry/CommonTopologies/src/GEMStripTopology.cc

```diff
@@ -0,0 +1,125 @@
+  const float  // y = (L/cos(phi))*mp.y()*cos(phi)
+      y(mp.y() * detHeight() + yCentreOfStripPlane()),
+      x(yAxisOrientation() * yDistanceToIntersection(y) * std::tan(stripAngle(mp.x())));
+  return LocalPoint(x, y);
+}
+
+LocalError GEMStripTopology::localError(float strip, float stripErr2) const {
+  const double phi(stripAngle(strip)), t1(std::tan(phi)), t2(t1 * t1),
+      tt(stripErr2 * std::pow(centreToIntersection() * angularWidth(), 2.f)),  // tangential sigma^2   *c2
+      rr(std::pow(detHeight(), 2.f) * (1.f / 12.f)),  // radial sigma^2( uniform prob density along strip)  *c2
+
+      xx(tt + t2 * rr), yy(t2 * tt + rr), xy(t1 * (rr - tt));
+
+  return LocalError(xx, xy, yy);
+}
+
+LocalError GEMStripTopology::localError(const MeasurementPoint& mp, const MeasurementError& me) const {
+  const double phi(stripAngle(mp.x())), s1(std::sin(phi)), c1(std::cos(phi)), cs(s1 * c1), s2(s1 * s1),
+      c2(1 - s2),  // rotation matrix
+
+      T(angularWidth() * (centreToIntersection() + yAxisOrientation() * mp.y() * detHeight()) /
+        c1),                // tangential measurement unit (local pitch)
+      R(detHeight() / c1),  // radial measurement unit (strip length)
+      tt(me.uu() * T * T),  // tangential sigma^2
+      rr(me.vv() * R * R),  // radial sigma^2
+      tr(me.uv() * T * R),
+
+      xx(c2 * tt + 2 * cs * tr + s2 * rr), yy(s2 * tt - 2 * cs * tr + c2 * rr), xy(cs * (rr - tt) + tr * (c2 - s2));
+
+  return LocalError(xx, xy, yy);
+}
+
+float GEMStripTopology::strip(const LocalPoint& lp) const {
+  const float  // phi is measured from y axis --> sign of angle is sign of x * yAxisOrientation --> use atan2(x,y), not atan2(y,x)
+      phi(std::atan2(lp.x(), yDistanceToIntersection(lp.y()))),
+      aStrip((phi - yAxisOrientation() * phiOfOneEdge()) / angularWidth());
+  return std::max(float(0), std::min((float)nstrips(), aStrip));
+}
+
+int GEMStripTopology::nearestStrip(const LocalPoint& lp) const {
+  return std::min(nstrips(), static_cast<int>(std::max(float(0), strip(lp))) + 1);
+}
+
+MeasurementPoint GEMStripTopology::measurementPosition(const LocalPoint& lp) const {
+  const float  // phi is [pi/2 - conventional local phi], use atan2(x,y) rather than atan2(y,x)
+      phi(yAxisOrientation() * std::atan2(lp.x(), yDistanceToIntersection(lp.y())));
+  return MeasurementPoint(yAxisOrientation() * (phi - phiOfOneEdge()) / angularWidth(),
+                          (lp.y() - yCentreOfStripPlane()) / detHeight());
+}
+
+MeasurementError GEMStripTopology::measurementError(const LocalPoint& p, const LocalError& e) const {
+  const double yHitToInter(yDistanceToIntersection(p.y())),
+      t(yAxisOrientation() * p.x() / yHitToInter),  // tan(strip angle)
+      cs(t / (1 + t * t)), s2(t * cs), c2(1 - s2),  // rotation matrix
+
+      T2(1. / (std::pow(angularWidth(), 2.f) *
+               (std::pow(p.x(), 2.f) + std::pow(yHitToInter, 2)))),  // 1./tangential measurement unit (local pitch) ^2
```

Comment:
let us avoid type conversion and use std::pow(p.x(), 2), 2.f is not needed here.